lib/network-layer/src/Cardano/Wallet/Network/Implementation/Ouroboros.hs

```diff
@@ -266,29 +266,34 @@ chainSyncFetchNextBlock queue =
 
     -- Ask the node whether it has the block.
     clientStIntersect
-        :: ClientStIntersect block point tip m Void
-    clientStIntersect =
+        :: point -> ClientStIntersect block point tip m Void
+    clientStIntersect point =
         ClientStIntersect
             { recvMsgIntersectFound = \_ _ ->
-                ChainSyncClient $ pure clientRequestBlock
+                ChainSyncClient $ pure $ clientRequestBlock point
             , recvMsgIntersectNotFound = \_ ->
                 result Nothing
             }
 
     -- Request the block in case of success.
     clientRequestBlock
-        :: ClientStIdle block point tip m Void
-    clientRequestBlock =
-        SendMsgRequestNext clientStNext (pure clientStNext)
+        :: point -> ClientStIdle block point tip m Void
+    clientRequestBlock point =
+        SendMsgRequestNext (clientStNext point) (pure $ clientStNext point)
 
     -- Fetch the block on rollforward.
     clientStNext
-        :: ClientStNext block point tip m Void
-    clientStNext = ClientStNext
+        :: point -> ClientStNext block point tip m Void
+    clientStNext point = ClientStNext
```

Comment:
ehh, this `point` is traveling a lot downwards ;-)

Reply:
🛷